internal/daemon/dpudaemon.go

```diff
@@ -207,6 +209,7 @@ func (d *DpuDaemon) Serve(listener net.Listener) error {
 
 	d.cancelManager = cancelManager
 
+	d.doneServing <- struct{}{}
```

Comment:
This wont work. We need to use the other done channel since this doesn't handle an error in other goroutines.

Reply:
Done as discussed.